lib/os/heap.c

```diff
@@ -220,6 +225,33 @@ static chunkid_t alloc_chunk(struct z_heap *h, size_t sz)
 	return 0;
 }
 
+/* O(n) with chunks in the top bucket.*/
+static void update_top_available(struct z_heap *h)
```

Comment:
Performance cost isn't high, but it's being taken on every allocation!  Lazy-evaluate this when the user requests the stats?